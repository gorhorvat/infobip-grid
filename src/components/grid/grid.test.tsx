import React from "react";
import { shallow } from 'enzyme';
import Grid from "./grid";
import toJson from 'enzyme-to-json';
import GridCell from "../grid-cell/grid-cell";

describe('Grid', () => {
  function setup(columns: number, rows: number) {
    return shallow(<Grid columns={columns} rows={rows} size="small" />);
  }

  it('Renders without crashing', () => {
    // Arrange
    const wrapper = setup(1, 1);

    // Assert
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('When grid cell is clicked its cell props should change', () => {
    // Arrange
    const wrapper = setup(1, 1);
    const cell = wrapper.find(GridCell);

    // Act
    cell.props().onCellClick();

    // Assert
    expect(cell.props().cell.isClicked).toBe(true);
  });

  it('When max number of cells is passed should render max number of cells (99)', () => {
    // Arrange
    const wrapper = setup(10, 10);

    // Assert
    expect(wrapper.find(GridCell).length).toBe(99);
  });
});