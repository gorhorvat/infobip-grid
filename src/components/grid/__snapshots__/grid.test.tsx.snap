// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Renders without crashing 1`] = `
<div
  className="grid grid--small"
>
  <table>
    <tbody>
      <tr
        key="0"
      >
        <GridCell
          cell={
            Object {
              "id": 0,
              "isClicked": false,
              "value": 1,
            }
          }
          key="0"
          onCellClick={[Function]}
        />
      </tr>
    </tbody>
  </table>
</div>
`;
